pkg/domain/service/place/service.go

```diff
@@ -0,0 +1,136 @@
+package place
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+	placeEntity "wantum/pkg/domain/entity/place"
+	"wantum/pkg/domain/repository"
+	"wantum/pkg/domain/repository/place"
+	"wantum/pkg/werrors"
+
+	"google.golang.org/grpc/codes"
+)
+
+type Service interface {
+	Create(ctx context.Context, masterTx repository.MasterTx, name string) (*placeEntity.Entity, error)
+	Update(ctx context.Context, masterTx repository.MasterTx, placeID int, name string) (*placeEntity.Entity, error)
+
+	Delete(ctx context.Context, masterTx repository.MasterTx, placeID int) error
+	UpDeleteFlag(ctx context.Context, masterTx repository.MasterTx, placeID int) (*placeEntity.Entity, error)
+	DownDeleteFlag(ctx context.Context, masterTx repository.MasterTx, placeID int) (*placeEntity.Entity, error)
+
+	GetByID(ctx context.Context, masterTx repository.MasterTx, placeID int) (*placeEntity.Entity, error)
+	GetAll(ctx context.Context, masterTx repository.MasterTx) (placeEntity.EntitySlice, error)
+}
+
+type service struct {
+	placeRepository place.Repository
+}
+
+func New(repo place.Repository) Service {
+	return &service{
+		placeRepository: repo,
+	}
+}
+
+func (s *service) Create(ctx context.Context, masterTx repository.MasterTx, name string) (*placeEntity.Entity, error) {
+	createdAt := time.Now()
+	place := &placeEntity.Entity{
+		Name:      name,
+		CreatedAt: &createdAt,
+		UpdatedAt: &createdAt,
+	}
+	result, err := s.placeRepository.Insert(ctx, masterTx, place)
+	if err != nil {
+		return nil, werrors.Stack(err)
+	}
+	place.ID = result
+	return place, nil
+}
+
+// WARNING: 空値があった時、元データが消滅する。
+// QUESTION: リクエストは、全フィールド埋める or 差分だけ
+func (s *service) Update(ctx context.Context, masterTx repository.MasterTx, placeID int, name string) (*placeEntity.Entity, error) {
+	place, err := s.placeRepository.SelectByID(ctx, masterTx, placeID)
+	if err != nil {
+		return nil, werrors.Stack(err)
+	}
+
+	updatedAt := time.Now()
+	place.Name = name
+	place.UpdatedAt = &updatedAt
+	err = s.placeRepository.Update(ctx, masterTx, place)
+	if err != nil {
+		return nil, werrors.Stack(err)
+	}
```

Comment:
[MUST]返り値が1つ（errorのみ）の場合は、ワンライナーで書こう！

```suggestion
	if err := s.placeRepository.Update(ctx, masterTx, place); err != nil {
		return nil, werrors.Stack(err)
	}
```